js/src/data/reducer.js

```diff
@@ -116,141 +117,76 @@ const reducer = ( state = DEFAULT_STATE, action ) => {
 
-			nextState.mc.shipping.times = times;
-			return nextState;
+			return set( state, 'mc.shipping.times', times );
 		}
 
 		case TYPES.DELETE_SHIPPING_TIMES: {
-			const nextState = getNextStateForShipping( state );
 			const countryCodeSet = new Set( action.countryCodes );
 			const times = state.mc.shipping.times.filter(
 				( el ) => ! countryCodeSet.has( el.countryCode )
 			);
 
-			nextState.mc.shipping.times = times;
-			return nextState;
+			return set( state, 'mc.shipping.times', times );
 		}
 
 		case TYPES.RECEIVE_SETTINGS:
 		case TYPES.SAVE_SETTINGS: {
-			return {
-				...state,
-				mc: {
-					...state.mc,
-					settings: {
-						...state.mc.settings,
-						...action.settings,
-					},
-				},
-			};
+			return set( state, 'mc.settings', action.settings );
 		}
 
 		case TYPES.RECEIVE_ACCOUNTS_JETPACK: {
-			const { account } = action;
-			const newState = cloneDeep( state );
-			newState.mc.accounts.jetpack = account;
-			return newState;
+			return set( state, 'mc.accounts.jetpack', action.account );
 		}
 
 		case TYPES.RECEIVE_ACCOUNTS_GOOGLE: {
-			const { account } = action;
-			const newState = cloneDeep( state );
-			newState.mc.accounts.google = account;
-			return newState;
+			return set( state, 'mc.accounts.google', action.account );
 		}
 
 		case TYPES.RECEIVE_ACCOUNTS_GOOGLE_ACCESS: {
-			return {
-				...state,
-				mc: {
-					...state.mc,
-					accounts: {
-						...state.mc.accounts,
-						google_access: action.data,
-					},
-				},
-			};
+			return set( state, 'mc.accounts.google_access', action.data );
 		}
 
 		case TYPES.RECEIVE_ACCOUNTS_GOOGLE_MC: {
-			const { account } = action;
-			const newState = cloneDeep( state );
-			newState.mc.accounts.mc = account;
-			return newState;
+			return set( state, 'mc.accounts.mc', action.account );
 		}
 
 		case TYPES.RECEIVE_ACCOUNTS_GOOGLE_MC_EXISTING: {
-			const { accounts } = action;
-			const newState = cloneDeep( state );
-			newState.mc.accounts.existing_mc = accounts;
-			return newState;
+			return set( state, 'mc.accounts.existing_mc', action.accounts );
 		}
 
 		case TYPES.RECEIVE_ACCOUNTS_GOOGLE_ADS: {
-			const { account } = action;
-			const newState = cloneDeep( state );
-			newState.mc.accounts.ads = account;
-			return newState;
+			return set( state, 'mc.accounts.ads', action.account );
 		}
 
 		case TYPES.DISCONNECT_ACCOUNTS_GOOGLE_ADS: {
-			return {
-				...state,
-				mc: {
-					...state.mc,
-					accounts: {
-						...state.mc.accounts,
-						ads: DEFAULT_STATE.mc.accounts.ads,
-					},
-				},
-			};
+			const value = DEFAULT_STATE.mc.accounts.ads;
+			return set( state, 'mc.accounts.ads', value );
 		}
 
 		case TYPES.RECEIVE_ACCOUNTS_GOOGLE_ADS_BILLING_STATUS: {
-			const { billingStatus } = action;
-			const newState = cloneDeep( state );
-			newState.mc.accounts.ads_billing_status = billingStatus;
-			return newState;
+			const path = 'mc.accounts.ads_billing_status';
+			return set( state, path, action.billingStatus );
```

Comment:
💅   we could set the path straight into the set function arguments, so it follows the same structure as the others. 

`return set( state,'mc.accounts.ads_billing_status', action.billingStatus );`

Reply:
Since `prettier` will ask for the below format, I tried to maintain visual consistency from the other perspective, so I wrote it in two lines.

```js
return set(
	state,
	'mc.accounts.ads_billing_status',
	action.billingStatus
);
```

Changed by https://github.com/woocommerce/google-listings-and-ads/pull/1151/commits/2a8f6af673e3b5f45f248d221c65d8032fb2f2f6